int32 save_path_flag
---
bool success_flag